src/core/MusicAlbum.ts

```diff
@@ -0,0 +1,67 @@
+import { Project } from './Project'
+import { BACKBLAZE_BUCKET } from './website_constants'
+
+export interface TrackDescriptor {
+  filename: string
+  title: string
+  date: string
+  description: string
+}
+
+export interface MusicAlbumDescriptor {
+  id: string
+  title: string
+  release_date: string
+  sort_key: string
+  description: string
+  tracks: TrackDescriptor[]
+  // Whether the track should advance or loop when it reaches the end.
+  play_style: 'advance' | 'loop'
+  hide?: boolean
+  featured?: boolean
+}
+
+export class MusicAlbum extends Project {
+  tracks: TrackDescriptor[]
+  play_style: 'advance' | 'loop'
+
+  private album_url: string
+
+  constructor(descriptor: MusicAlbumDescriptor) {
+    super({
+      id: descriptor.id,
+      title: `${descriptor.title} 🎵`,
+      years: descriptor.release_date,
+      sort_key: descriptor.sort_key,
+      description: descriptor.description,
+      img_format: 'png',
+      featured: descriptor.featured,
+    })
+
+    this.url = `/album/${this.id}`
+    this.album_url = `${BACKBLAZE_BUCKET}/music-albums/${this.id}`
+    const thumbnail_url = `${BACKBLAZE_BUCKET}/project-thumbnails/${this.id}.png`
```

Comment:
If this was a getter on Project, we could override it and avoid having to explicitly set the thumbnail in the subclass.